### PR TITLE
Improvements to building Docker container and smoke tests

### PR DESCRIPTION
**Pull request by eproxus**
Also ignores crash dumps in the root directory.

If you can come up with other ideas of things that can be ignored safely, I can amend the PR.

### CODE REVIEW

**Comment by eproxus** [comment]
@lucafavatella I saw it in the `.gitignore` file, but since I didn't have it I didn't want to add it

File: Makefile [hunk 1]
```diff
@@ -162,7 +162,13 @@ all-tests: eunit test
 docker:
 	@docker build . -t aeternity/epoch:local
 
-smoke-test: system-test-deps docker
+docker-clean:
+	@docker image prune -f
+	@docker container prune -f
```
Comment (lucafavatella) [inline Makefile]: This removes stopped containers.

File: Makefile [hunk 2]
```diff
@@ -162,7 +162,13 @@ all-tests: eunit test
 docker:
 	@docker build . -t aeternity/epoch:local
 
-smoke-test: system-test-deps docker
+docker-clean:
+	@docker image prune -f
```
Comment (lucafavatella) [inline Makefile]: This removes unused images.

Comment (sylane) [inline Makefile]: You may want to do the following to cleanup any system test containers and network:
```
docker rm -f $(docker ps -a | grep 'aeternity/epoch' | cut -d ' ' -f 1)
docker network prune -f
```

File: Makefile (hunk 1 above)
Comment (eproxus) [reply Makefile]: This might be too much? The target has to be run manually though...

**Comment by dincho** [comment]
> Some kind of filter of what images and containers being deleted would be very helpful.

👍

**Comment by eproxus** [comment]
@dincho I'll update with filters.

@lucafavatella Why would it differ if the order was different?

**Comment by lucafavatella** [comment]
> @lucafavatella Why would it differ if the order was different?

@eproxus Because I expect that a stopped container still makes its image count as used.

**Comment by eproxus** [comment]
@dincho @lucafavatella New strategy: I'm adding a tag to all containers and images created and only cleaning those

**Pull request merged**